### ISSUE TITLE
EPI2ME-lab launcher proxy

### ISSUE THREAD

**Issue by chienchi** [epi2me-labs/labslauncher#5]
I would like to use the EPI2ME-lab launcher in my Macbook pro which is current has MacOS Catalina (version 10.5.7). When I click on the EPI2ME-Labs-Launcher v1.0.0 downloaded from the github site, I have the following error.  I think it could be my network is behide the corporation proxy.  But I am not sure how do I solve this issue.  

For making sure that proxy could be an issue, I bring my laptop home and using my home internet.  The  EPI2ME-lab launcher works.  

Is it possible to pass proxy info or use the system setting when using the EPI2ME launcher?

Thank you,

Chienchi

### labslauncher.log 
```
[16:56:18 - labslauncher] Starting application.
[16:56:18 - DckrClnt] Creating docker client with options:
           image name: ontresearch/epi2melabs-notebook
           server name: Epi2Me-Labs-Server
           data bind: /epi2melabs/
           command: start-notebook.sh  --NotebookApp.allow_origin='https://colab.research.google.com' --NotebookApp.disable_check_xsrf=True --NotebookApp.port_retries=0 --no-browser --notebook-dir=/
           host only: True
           fixed tag: None
[16:56:18 - DckrClnt] Connection to docker (re)established.
[16:56:18 - labslauncher] Uncaught exception:
Traceback (most recent call last):
  File "urllib3/connection.py", line 159, in _new_conn
  File "urllib3/util/connection.py", line 80, in create_connection
  File "urllib3/util/connection.py", line 70, in create_connection
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 600, in urlopen
  File "urllib3/connectionpool.py", line 343, in _make_request
  File "urllib3/connectionpool.py", line 839, in _validate_conn
  File "urllib3/connection.py", line 301, in connect
  File "urllib3/connection.py", line 168, in _new_conn
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x124d25048>: Failed to establish a new connection: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 449, in send
  File "urllib3/connectionpool.py", line 638, in urlopen
  File "urllib3/util/retry.py", line 399, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='hub.docker.com', port=443): Max retries exceeded with url: /v2/repositories/ontresearch/epi2melabs-notebook/tags (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x124d25048>: Failed to establish a new connection: [Errno 61] Connection refused',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "build/EPI2ME-Labs-Launcher/labslauncher-script.py", line 2, in <module>
  File "labslauncher/app.py", line 825, in main
  File "labslauncher/app.py", line 525, in __init__
  File "labslauncher/dockerutil.py", line 176, in __init__
  File "labslauncher/dockerutil.py", line 216, in is_running
  File "labslauncher/dockerutil.py", line 235, in latest_available_tag
  File "labslauncher/dockerutil.py", line 92, in newest_tag
  File "labslauncher/dockerutil.py", line 47, in get_image_tags
  File "cachetools/decorators.py", line 22, in wrapper
  File "labslauncher/dockerutil.py", line 29, in _get_image_meta
  File "requests/api.py", line 75, in get
  File "requests/api.py", line 60, in request
  File "requests/sessions.py", line 524, in request
  File "requests/sessions.py", line 637, in send
  File "requests/adapters.py", line 516, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='hub.docker.com', port=443): Max retries exceeded with url: /v2/repositories/ontresearch/epi2melabs-notebook/tags (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x124d25048>: Failed to establish a new connection: [Errno 61] Connection refused',))
 ```

**Comment by cjw85** [comment]
Hi @chienchi 

Allowing the launcher to use a proxy is something we are going to look at for the next release. The launcher makes a few requests to dockerhub in order to be notified about updates and to download those updates.

**Comment by cjw85** [comment]
I've just uploaded version 1.0.2, this has the ability to specify an https proxy in the Settings (Preferences on macOS) menu of the application. You will also need to set the proxy settings in Docker Desktop/Preferences/Resource/PROXIES menu.

**Comment by chienchi** [comment]
Thanks.  I downloaded version 1.0.2 and confirmed the proxy setup and the application starts when setup the Settings (Preferences on macOS) menu of the application and docker preference.  Thumb up for the quick update.   

One small thing that it shows up there is an update available for notebook server v1.0.3 but the update button doesn't response.
